### ISSUE TITLE
RBM: The performance of pretraining is not good, when using use Gaussian units model.

### ISSUE THREAD

**Issue by rosefun** [blackecho/Deep-Learning-TensorFlow#72]
As my dataset is continous variables, I set the 'rbm_gauss_visible' True. 

However, I find that the performance of the model with pretraining is the same as without pretraining.
I think the ability of RBM in extracting the features is not good enough.

The pretrain reconstruction loss is as follows:
```

Reconstruction loss: 1.3044015:   0% 1/400 [00:00<02:32, 
2.62it/s]Reconstruction loss: 1.2943473:   0%| 1/400 [00:00<03:24,  
1.95it/s]Reconstruction loss: 1.2943473:   0% | 2/400 [00:00<01:42,  
3.90it/s]Reconstruction loss: 1.2825284:   0% | 2/400 [00:00<02:07, 
 3.12it/s]Reconstruction loss: 1.2825284:   1%| | 3/400 [00:00<01:24,  
4.67it/s]Reconstruction loss: 1.2711711:   1%|▍                                                 | 3/400 [00:00<01:43,  3.82it/s]Reconstruction loss: 1.2711711:   1%|▌                                                 | 4/400 [00:00<01:17,  5.09it/s]Reconstruction loss: 1.2646809:   1%|▌                                                 | 4/400 [00:00<01:31,  4.33it/s]Reconstruction loss: 1.2646809:   1%|▋                                                 | 5/400 [00:00<01:12,  5.42it/s]Reconstruction loss: 1.2509606:   1%|▋                                                 | 5/400 [00:01<01:24,  4.69it/s]Reconstruction loss: 1.2509606:   2%|▊                                                 | 6/400 [00:01<01:10,  5.63it/s]Reconstruction loss: 1.2504419:   2%|▊                                                 | 6/400 [00:01<01:18,  5.00it/s]Reconstruction loss: 1.2504419:   2%|▉                                                 | 7/400 [00:01<01:07,  5.84it/s]Reconstruction loss: 1.2381066:   2%|▉                                                 | 7/400 [00:01<01:14,  5.26it/s]Reconstruction loss: 1.2381066:   2%|█                                                 | 8/400 [00:01<01:05,  6.01it/s]Reconstruction loss: 1.2347949:   2%|█                                                 | 8/400 [00:01<01:12,  5.43it/s]Reconstruction loss: 1.2347949:   2%|█▏                                                | 9/400 [00:01<01:04,  6.11it/sReconstruction loss: 1.2286505:   2%|█▏                                                | 9/400 [00:01<01:10,  5.58it/sReconstruction loss: 1.2286505:   2%|█▏                                               | 10/400 [00:01<01:02,  6.20it/sReconstruction loss: 1.2220185:   2%|█▏                                               | 10/400 [00:01<01:08,  5.71it/sReconstruction loss: 1.2220185:   3%|█▎                                               | 11/400 [00:01<01:01,  6.28it/sReconstruction loss: 1.2170618:   3%|█▎                                               | 11/400 [00:01<01:06,  5.84it/sReconstruction loss: 1.2170618:   3%|█▍                                               | 12/400 [00:01<01:00,  6.37it/sReconstruction loss: 1.2137262:   3%|█▍                                               | 12/400 [00:02<01:05,  5.96it/sReconstruction loss: 1.2137262:   3%|█▌                                               | 13/400 [00:02<00:59,  6.45it/sReconstruction loss: 1.2091556:   3%|█▌                                               | 13/400 [00:02<01:03,  6.05it/sReconstruction loss: 1.2091556:   4%|█▋                                               | 14/400 [00:02<00:59,  6.51it/sReconstruction loss: 1.2046573:   4%|█▋                                               | 14/400 [00:02<01:02,  6.15it/sReconstruction loss: 1.2046573:   4%|█▊                                               | 15/400 [00:02<00:58,  6.58it/sReconstruction loss: 1.208482:   4%|█▉                                                | 15/400 [00:02<01:02,  6.21it/sReconstruction loss: 1.208482:   4%|██                                                | 16/400 [00:02<00:57,  6.62it/sReconstruction loss: 1.2066234:   4%|█▉                                               | 16/400 [00:02<01:01,  6.27it/sReconstruction loss: 1.2066234:   4%|██                                               | 17/400 [00:02<00:57,  6.66it/sReconstruction loss: 1.2018061:   4%|██                                               | 17/400 [00:02<01:00,  6.29it/sReconstruction loss: 1.2018061:   4%|██▏                                              | 18/400 [00:02<00:57,  6.65it/Reconstruction loss: 1.2018793:   4%|██▏                                              | 18/400 [00:02<01:00,  6.29it/Reconstruction loss: 1.2018793:   5%|██▎                                              | 19/400 [00:02<00:57,  6.64it/Reconstruction loss: 1.2035196:   5%|██▎                                              | 19/400 [00:03<01:00,  6.32it/Reconstruction loss: 1.2035196:   5%|██▍                                              | 20/400 [00:03<00:57,  


```

```

Reconstruction loss: 1.1740133:  99%|
Reconstruction loss: 1.1779407:  99%
 Reconstruction loss: 1.1779407:  99%|
Reconstruction loss: 1.1700155:  99%|
Reconstruction loss: 1.1700155: 100%
Reconstruction loss: 1.176933: 100%|
Reconstruction loss: 1.176933: 100%|
Reconstruction loss: 1.1773719: 100%
Reconstruction loss: 1.1773719: 100%|

```

No matter the model is trained with pretraining phase or not, the final result is almost the same about 0.83.

Anyone knows the problem?
